### PR TITLE
Add ToggleGroup component

### PR DESCRIPTION
**Pull request by Diogomartf**
Based on https://headlessui.com/react/radio-group

image

### CODE REVIEW

File: ui/ToggleGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+export const toggleOptionStyles = cva(
+  [
+    "flex items-center py-2 font-medium leading-5 rounded-12",
+    "outline-none cursor-pointer",
+  ],
+  {
+    variants: {
+      active: {
+        true: ["bg-surface-surface-0 shadow-3"],
```
Comment (berteotti) [inline ui/ToggleGroup.tsx]: Shadow is too big. Should be `shadow-2`

File: ui/ToggleGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    <RadioGroup
+      value={value}
+      onChange={onChange}
+      className="flex space-x-1 bg-surface-surface-2 p-[4px] rounded-12 w-fit"
+      {...props}
+    >
+      {renderChildrenWithProps()}
+    </RadioGroup>
+  );
+};
```
Comment (berteotti) [inline ui/ToggleGroup.tsx]: - We are doing too complex things here. Let's keep it simple. This component should only render `{children}`, if they have the wrong size props that's a dev / design problem. The dev should specify the correct button sizes.

Comment (Diogomartf) [reply ui/ToggleGroup.tsx]: Ok, let's simplify for now.

File: app/ui/page.tsx [hunk 3]
```diff
@@ -317,7 +325,49 @@ export default function UI() {
...
+                    auto
+                  </ToogleGroupOption>
+                  <ToogleGroupOption
+                    size={size as ToogleOptionSizeProp}
+                    value="0.1"
+                  >
+                    0.1%
+                  </ToogleGroupOption>
+                  <ToogleGroupOption
+                    size={size as ToogleOptionSizeProp}
```
Comment (berteotti) [inline app/ui/page.tsx]: If we are casting types here, there is something wrong in our types.

Comment (berteotti) [reply app/ui/page.tsx]: if we add  types to `["xs", "sm", "md", "lg"]` we don't need to cast. So we just declare the type in 1 place

File: ui/ToggleGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  onChange: Dispatch<SetStateAction<string>>;
+  children: React.ReactNode;
+}
+
+export const ToggleGroup = ({
+  value,
+  onChange,
+  children,
+  ...props
+}: ToggleGroupProp) => {
```
Comment (berteotti) [inline ui/ToggleGroup.tsx]:
```suggestion
export const ToggleGroup = (
  props: RadioGroupProps<React.ElementType, string>
) => {
```
We can use this types here, and we just need to spread props to `RadioGroup`

File: ui/ToggleGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  children: ReactNode;
+  size?: ToogleOptionSizeProp;
+  value: string;
+}
+export const ToogleGroupOption = ({
+  children,
+  value,
+  size,
+  ...props
+}: ToggleOptionProp) => {
```
Comment (berteotti) [inline ui/ToggleGroup.tsx]:
```suggestion
type ToggleOptionProp = RadioOptionProps<React.ElementType, string> &
  React.PropsWithChildren<{
    size?: ToogleOptionSizeProp;
  }>;

export const ToogleGroupOption = ({
  children,
  size,
  ...props
}: ToggleOptionProp) => {
```

We can use this types and props here. Value can be passed inside `props`

**Review by berteotti — APPROVED** [review]
Looking goooood

**Pull request merged**